03-JavaServiceSkeleton

d-serj opened the pull request:
Impelented simple service to store string. Implemented 2 applications to receive and transmit string from/to the service

Commit:
03-JavaServiceSkeleton: Add homework

Task is described in the README.md file.

BulldozerBSG committed:
03-JavaServiceSkeleton: Update README.md

Layout correction

d-serj committed:
03-JavaServiceSkeleton: Add string service

Add string service which can receive and transmit strings

d-serj committed:
03-JavaServiceSkeleton: Add String transmitter application

Add application which can transmit the entered string to the String Service

Signed-off-by: d-serj <serj.dub1na@gmail.com>

d-serj committed:
03-JavaServiceSkeleton: Add String transmitter application

Add application which can receive string from the String Service
and show it on the screen

Signed-off-by: d-serj <serj.dub1na@gmail.com>

d-serj committed:
03-JavaServiceSkeleton: Fix button ID

Fix transmit string button ID

Signed-off-by: d-serj <serj.dub1na@gmail.com>

BulldozerBSG commented on 03-JavaServiceSkeleton/StringsReceiver/app/src/main/java/com/example/stringsreceiver/MainActivity.java:
```diff
@@ -0,0 +1,61 @@
+
+        Intent mIntent = new Intent();
+        mIntent.setClassName("com.example.stringservice", "com.example.stringservice.StringService");
+        bindService(mIntent, mConnection, BIND_AUTO_CREATE);
```
Where is unbindService() ?

d-serj replied:
unbindService() was added

BulldozerBSG commented on 03-JavaServiceSkeleton/StringsReceiver/app/src/main/java/com/example/stringsreceiver/MainActivity.java:
```diff
@@ -0,0 +1,61 @@
+                try {
+                    output = mService.stringReceive();
+                    strOutput.setText(output);
+                } catch (StringIndexOutOfBoundsException | RemoteException | NullPointerException e) {
```
What is StringIndexOutOfBoundsException for here?

d-serj replied:
Removed

BulldozerBSG commented on 03-JavaServiceSkeleton/StringsTransmitter/app/src/main/java/com/example/stringstransmitter/MainActivity.java:
```diff
@@ -0,0 +1,60 @@
+
+                try {
+                    mService.stringTransmit(input);
+                } catch (StringIndexOutOfBoundsException | RemoteException | NullPointerException e) {
```
What is StringIndexOutOfBoundsException for here?

d-serj replied:
Removed

BulldozerBSG commented on 03-JavaServiceSkeleton/StringsTransmitter/app/src/main/java/com/example/stringstransmitter/MainActivity.java:
```diff
@@ -0,0 +1,60 @@
+
+        Intent mIntent = new Intent();
+        mIntent.setClassName("com.example.stringservice", "com.example.stringservice.StringService");
+        bindService(mIntent, mConnection, BIND_AUTO_CREATE);
```
Where is unbindService() ?

d-serj replied:
unbindService() was added

d-serj committed:
03-JavaServiceSkeleton: Remove unnecessary exception condition

Remove StringIndexOutOfBoundsException exception

Signed-off-by: d-serj <serj.dub1na@gmail.com>

d-serj committed:
03-JavaServiceSkeleton: Add unbindService() call to StringsTransmitter app

Add unbindService() call in order to destroy service object on application exit

Signed-off-by: d-serj <serj.dub1na@gmail.com>

d-serj committed:
03-JavaServiceSkeleton: Add unbindService() call to StringsReceiver app

Add unbindService() call in order to destroy service object on application exit

Signed-off-by: d-serj <serj.dub1na@gmail.com>

d-serj committed:
03-JavaServiceSkeleton: Add exception handler for bindService call

Signed-off-by: d-serj <serj.dub1na@gmail.com>
